src/main/webapp/app/exercises/programming/shared/code-editor/container/code-editor-container.component.ts

```diff
@@ -216,10 +217,18 @@ export class CodeEditorContainerComponent implements ComponentCanDeactivate {
     /**
      * Show an error as an alert in the top of the editor html.
      * Used by other components to display errors.
-     * The error must already be provided translated by the emitting component.
+     * @param error the translation key of the error that should be displayed
      */
     onError(error: any) {
-        this.alertService.error(`artemisApp.editor.errors.${error as string}`);
+        let errorTranslationKey: string;
+        const translationParams = { connectionIssue: '' };
+        if (typeof error !== 'string' || !error.includes(ConnectionError.message)) {
+            errorTranslationKey = error as string;
```

Comment:
Is this correct? You call `error as string` explicitly in cases where the error is not if type string.

Reply:
Typecheck now removed. We also assumed before that the error is a string